packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -123,66 +131,170 @@ contract L2Genesis is Script, Artifacts {
+        setEAS(); // 21
+        setGovernanceToken(); // 42: OP (not behind a proxy)
+    }
+
+    function setProxyAdmin() public {
+        // Note the ProxyAdmin implementation itself is behind a proxy that owns itself.
+        _setImplementationCode(Predeploys.PROXY_ADMIN);
+
+        bytes32 _ownerSlot = bytes32(0);
+
+        // there is no initialize() function, so we just set the storage manually.
+        vm.store(Predeploys.PROXY_ADMIN, _ownerSlot, bytes32(uint256(uint160(cfg.proxyAdminOwner()))));
+    }
+
+    function setL2ToL1MessagePasser() public {
+        _setImplementationCode(Predeploys.L2_TO_L1_MESSAGE_PASSER);
+    }
+
+    /// @notice This predeploy is following the safety invariant #1.
+    function setL2CrossDomainMessenger(address payable _deployedL1CrossDomainMessengerProxy) public {
+        address impl = _setImplementationCode(Predeploys.L2_CROSS_DOMAIN_MESSENGER);
+
+        L2CrossDomainMessenger(impl).initialize({ _l1CrossDomainMessenger: L1CrossDomainMessenger(address(0)) });
+
+        L2CrossDomainMessenger(Predeploys.L2_CROSS_DOMAIN_MESSENGER).initialize({
+            _l1CrossDomainMessenger: L1CrossDomainMessenger(_deployedL1CrossDomainMessengerProxy)
+        });
+    }
+
+    /// @notice This predeploy is following the safety invariant #1.
+    function setL2StandardBridge(address payable _deployedL1StandardBridgeProxy) public {
+        address impl = _setImplementationCode(Predeploys.L2_STANDARD_BRIDGE);
+
+        L2StandardBridge(payable(impl)).initialize({ _otherBridge: L1StandardBridge(payable(address(0))) });
+
+        L2StandardBridge(payable(Predeploys.L2_STANDARD_BRIDGE)).initialize({
+            _otherBridge: L1StandardBridge(_deployedL1StandardBridgeProxy)
+        });
+    }
+
+    /// @notice This predeploy is following the safety invariant #1.
+    function setL2ERC721Bridge(address payable _deployedL1ERC721BridgeProxy) public {
+        address impl = _setImplementationCode(Predeploys.L2_ERC721_BRIDGE);
+
+        L2ERC721Bridge(impl).initialize({ _l1ERC721Bridge: payable(address(0)) });
+
+        L2ERC721Bridge(Predeploys.L2_ERC721_BRIDGE).initialize({ _l1ERC721Bridge: payable(_deployedL1ERC721BridgeProxy) });
+    }
+
+    /// @notice This predeploy is following the safety invariant #2,
+    function setSequencerFeeVault() public {
+        SequencerFeeVault vault = new SequencerFeeVault({
+            _recipient: cfg.sequencerFeeVaultRecipient(),
+            _minWithdrawalAmount: cfg.sequencerFeeVaultMinimumWithdrawalAmount(),
+            _withdrawalNetwork: FeeVault.WithdrawalNetwork(cfg.sequencerFeeVaultWithdrawalNetwork())
+        });
+
+        address impl = Predeploys.predeployToCodeNamespace(Predeploys.SEQUENCER_FEE_WALLET);
+        console.log("Setting %s implementation at: %s", "SequencerFeeVault", impl);
+        vm.etch(impl, address(vault).code);
+
+        /// Reset so its not included state dump
+        vm.etch(address(vault), "");
```

Comment:
it would probably be good to turn the reset into an internal function and reuse its impl everywhere. You can see how forge-std does it [here](https://github.com/foundry-rs/forge-std/blob/5dd1c68131ddd3c89ef169666eb262b92e90507c/src/StdCheats.sol#L547)

we don't need to go as far as they do, for us setting the balance to 0 technically isn't required as we don't have balance in this context

Reply:
This is not a full reset, but rather the reset of just the specific contract that was temporarily deployed for genesis purposes. Open to suggestions for how the reset should be, but don't think there's a lot to reduce.